hermes/bin/src/runtime_extensions/hermes/cardano/event.rs

```diff
@@ -1,20 +1,29 @@
 //! Cardano Blockchain runtime extension event handler implementation.
 
 use crate::{
-    event::HermesEventPayload,
-    runtime_extensions::bindings::hermes::cardano::api::{
-        BlockSrc, CardanoBlock, CardanoBlockchainId, CardanoTxn,
+    app::ApplicationName,
+    event::{HermesEvent, HermesEventPayload, TargetApp, TargetModule},
+    runtime_extensions::bindings::{
+        exports::hermes::cardano::event_on_block::{CardanoNetwork, SubscriptionId},
+        hermes::cardano::api::Slot,
     },
+    wasm::module::ModuleId,
 };
 
 /// On Cardano block event
 pub(super) struct OnCardanoBlockEvent {
-    /// The blockchain id the block originated from.
-    pub(super) blockchain: CardanoBlockchainId,
-    /// This raw CBOR block data.
-    pub(super) block: CardanoBlock,
-    /// Source information about where the block came from, and if we are at tip or not.
-    pub(super) source: BlockSrc,
+    /// The Cardano blockchain network that the block event is originated from.
+    network: CardanoNetwork,
+    /// A unique identifier of the block subscription.
+    subscription_id: SubscriptionId,
+    /// The Cardano slot number that the block event is in.
+    slot: Slot,
```

Comment:
see comment on wit PR.  I think this should just be the block resource, and all of this other data can be obtained directly from that resource, so it doesn't need to be in the event directly.